binheap.py

```diff
@@ -17,11 +17,10 @@ def push(self, value):
 
     def __parent(self, index):
```

Comment:
Yeah,  I figured.  

There is no such thing as "private" methods in Python OOP.  You can indicate that a method is intended to be a private API.  the standard is to prefix such names with a single underscore.  

The double-underscore prefix actually engages a Python system called *name mangling*.  What that means is that if the name of your method is `__foo` and the class is called `Bar`, your method is actually bound to the symbol `_bar_foo` in the class namespace.  The intention of this special behavior is to allow you to create methods that people who *subclass* your classes will not be able to easily override.  A side-effect is that the names of the methods are harder to guess, but it still doesn't actually make the methods "private" in any real sense.